python-avd/pyavd/_eos_designs/structured_config/flows/__init__.py

```diff
@@ -124,70 +110,95 @@ def _enable_sflow(self) -> bool:
+            sample=self.inputs.flow_tracking_settings.sampled.sample,
+            shutdown=False,
+            encapsulation=EosCliConfigGen.FlowTracking.Sampled.Encapsulation(
+                ipv4_ipv6=self.inputs.flow_tracking_settings.sampled.encapsulation.ipv4_ipv6,
+                mpls=self.inputs.flow_tracking_settings.sampled.encapsulation.mpls,
+            ),
+            hardware_offload=EosCliConfigGen.FlowTracking.Sampled.HardwareOffload(
+                ipv4=self.inputs.flow_tracking_settings.sampled.hardware_offload.ipv4,
+                ipv6=self.inputs.flow_tracking_settings.sampled.hardware_offload.ipv6,
+                threshold_minimum=self.inputs.flow_tracking_settings.sampled.hardware_offload.threshold_minimum,
+            ),
+        )
 
-        return flow_tracking
+        # Validate and configure trackers
+        default_tracker = next(iter(EosDesigns.FlowTrackingSettings().trackers))
+        for tracker_name in natural_sort(trackers):
+            config = self._get_tracker_input_config(default_tracker, tracker_name)
 
-    def _get_enabled_flow_trackers(self) -> set[str]:
-        """
-        Enable flow-tracking if any interface is enabled for flow-tracking.
+            # Need to handle mpls specifically
+            record_export = config.record_export._cast_as(EosCliConfigGen.FlowTracking.Sampled.TrackersItem.RecordExport)
+            record_export.mpls = config.sampled.record_export.mpls
 
-        This relies on flow-tracking being rendered after all other eos_designs modules (except structured config).
+            self.structured_config.flow_tracking.sampled.trackers.append_new(
+                name=config.name,
+                record_export=record_export,
+                exporters=config.exporters._cast_as(EosCliConfigGen.FlowTracking.Sampled.TrackersItem.Exporters),
+                table_size=config.sampled.table_size,
+            )
+
+    def _get_tracker_input_config(
+        self, default_tracker: EosDesigns.FlowTrackingSettings.TrackersItem, tracker_name: str
+    ) -> EosDesigns.FlowTrackingSettings.TrackersItem:
+        """Retrieves inputs for the given tracker_name.
+
+        If the name matches the default tracker_name, return the default tracker.
+
+        We allow overriding the default flow tracker name, so if user has configured a tracker
+        with the default tracker name, then we just use that, if not, we create a default config
         """
-        all_interfaces = chain(
-            self.structured_config.ethernet_interfaces, self.structured_config.port_channel_interfaces, self.structured_config.dps_interfaces
-        )
-        if self.shared_utils.flow_tracking_type == "hardware":
-            return {interface.flow_tracker.hardware for interface in all_interfaces if interface.flow_tracker.hardware}
+        if (tracker_settings := self.inputs.flow_tracking_settings.trackers.get(tracker_name)) != Undefined:
```

Comment:
IMO a `if tracker_name in self.inputs.flow_tracking_settings.trackers` is simpler. and then return the item with `[tracker_name]`. Two cheap operations and simpler to read.